src/code/z_map_disp.c

```diff
@@ -1,109 +1,1877 @@
+};
+static u32 sLoadTextureBlock_siz_INCR[4] = {
+    G_IM_SIZ_4b_INCR,
+    G_IM_SIZ_8b_INCR,
+    G_IM_SIZ_16b_INCR,
+    G_IM_SIZ_32b_INCR,
+};
+static s32 sLoadTextureBlock_siz_SHIFT[4] = {
+    G_IM_SIZ_4b_SHIFT,
+    G_IM_SIZ_8b_SHIFT,
+    G_IM_SIZ_16b_SHIFT,
+    G_IM_SIZ_32b_SHIFT,
+};
+static u32 sLoadTextureBlock_siz_BYTES[4] = {
+    G_IM_SIZ_4b_BYTES,
+    G_IM_SIZ_8b_BYTES,
+    G_IM_SIZ_16b_BYTES,
+    G_IM_SIZ_32b_BYTES,
+};
+static u32 sLoadTextureBlock_siz_LINE_BYTES[4] = {
+    G_IM_SIZ_4b_LINE_BYTES,
+    G_IM_SIZ_8b_LINE_BYTES,
+    G_IM_SIZ_16b_LINE_BYTES,
+    G_IM_SIZ_32b_LINE_BYTES,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_80102EA4.s")
+#define gDPLoadTextureBlock_TEST(pkt, timg, fmt, siz, width, height, pal, cms, cmt, masks, maskt, shifts, shiftt)      \
+    _DW({                                                                                                              \
+        gDPSetTextureImage(pkt, fmt, sLoadTextureBlock_siz_LOAD_BLOCK[siz], 1, timg);                                  \
+        gDPSetTile(pkt, fmt, sLoadTextureBlock_siz_LOAD_BLOCK[siz], 0, 0, G_TX_LOADTILE, 0, cmt, maskt, shiftt, cms,   \
+                   masks, shifts);                                                                                     \
+        gDPLoadSync(pkt);                                                                                              \
+        gDPLoadBlock(pkt, G_TX_LOADTILE, 0, 0,                                                                         \
+                     (((width) * (height) + sLoadTextureBlock_siz_INCR[siz]) >> sLoadTextureBlock_siz_SHIFT[siz]) - 1, \
+                     CALC_DXT(width, sLoadTextureBlock_siz_BYTES[siz]));                                               \
+        gDPPipeSync(pkt);                                                                                              \
+        gDPSetTile(pkt, fmt, sLoadTextureBlock_siz[siz], (((width)*sLoadTextureBlock_siz_LINE_BYTES[siz]) + 7) >> 3,   \
+                   0, G_TX_RENDERTILE, pal, cmt, maskt, shiftt, cms, masks, shifts);                                   \
+        gDPSetTileSize(pkt, G_TX_RENDERTILE, 0, 0, ((width)-1) << G_TEXTURE_IMAGE_FRAC,                                \
+                       ((height)-1) << G_TEXTURE_IMAGE_FRAC);                                                          \
+    })
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_80102EB4.s")
+static UNK_TYPE4 D_801BEB30[2] = { 0, 0 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_80102ED0.s")
+static u64 sWhiteSquareTex[] = {
+    0xFFFFFFFFFFFFFFFF, 0xFFFFFFFFFFFFFFFF, 0xFFFFFFFFFFFFFFFF, 0xFFFFFFFFFFFFFFFF,
+    0xFFFFFFFFFFFFFFFF, 0xFFFFFFFFFFFFFFFF, 0xFFFFFFFFFFFFFFFF, 0xFFFFFFFFFFFFFFFF,
+    0xFFFFFFFFFFFFFFFF, 0xFFFFFFFFFFFFFFFF, 0xFFFFFFFFFFFFFFFF, 0xFFFFFFFFFFFFFFFF,
+    0xFFFFFFFFFFFFFFFF, 0xFFFFFFFFFFFFFFFF, 0xFFFFFFFFFFFFFFFF, 0xFFFFFFFFFFFFFFFF,
+};
```

Comment:
I know it is just a simple white square, but I wonder if we should still extract this with ZAPD?

Reply:
Bump